scripts/kconfig/kconfigfunctions.py

```diff
@@ -0,0 +1,77 @@
+#
+# Copyright (c) 2018-2019 Linaro
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import sys
+import os
+
+# Types we support
+# 'string', 'int', 'hex', 'bool'
+conf_file = os.path.join(os.environ['PROJECT_BINARY_DIR'], "include/generated/generated_dts_board.conf")
+
+dt_defines = {}
+if os.path.isfile(conf_file):
+    for line in open(conf_file, 'r', encoding='utf-8'):
+        if '=' in line:
+            c = line.split('=')
+            dt_defines[c[0]] = c[1].strip()
+
+def _dt_units_to_scale(unit):
+    if not unit:
+        return 0
+    if unit == 'k' or unit == 'K':
+        return 10
+    if unit == 'm' or unit == 'M':
+        return 20
+    if unit == 'g' or unit == 'G':
+        return 30
+
+def dt_int_val(kconf, _, name, unit=None):
+    """
+    This function looks up 'name' in the DTS generated "conf" style database
+    and if its found it will return the value as an decimal integer.  The
+    function will scale the value based on 'unit':
+        'k' or 'K'  scale by 1024 (1 << 10)
+        'm' or 'M'  scale by 1,048,576 (1 << 20)
+        'g' or 'G'  scale by 1,073,741,824 (1 << 30)
+    """
+    if name not in dt_defines:
+        return "0"
+
+    d = dt_defines[name]
+    if d.startswith(('0x', '0X')):
+        d = int(d, 16)
+    else:
+        d = int(d)
+    d >>= _dt_units_to_scale(unit)
+
+    return str(d)
+
+def dt_hex_val(kconf, _, name, unit=None):
+    """
+    This function looks up 'name' in the DTS generated "conf" style database
+    and if its found it will return the value as an hex integer.  The
+    function will scale the value based on 'unit':
+        'k' or 'K'  scale by 1024 (1 << 10)
+        'm' or 'M'  scale by 1,048,576 (1 << 20)
+        'g' or 'G'  scale by 1,073,741,824 (1 << 30)
+    """
+    if name not in dt_defines:
+        return "0x0"
+
+    d = dt_defines[name]
+    if d.startswith(('0x', '0X')):
+        d = int(d, 16)
+    else:
+        d = int(d)
+    d >>= _dt_units_to_scale(unit)
+    d = hex(d)
+
+    return str(d)
```

Comment:
Could get rid of the `str()`. `hex()` returns a string already.

Reply:
Done